src/daft-logical-plan/src/ops/join.rs

```diff
@@ -106,24 +99,31 @@ impl Join {
 
     /// Add a project under the right side plan when necessary in order to resolve naming conflicts
     /// between left and right side columns.
-    #[allow(clippy::too_many_arguments)]
-    pub(crate) fn rename_right_columns(
-        left: LogicalPlanRef,
+    pub(crate) fn deduplicate_join_columns(
+        left: &LogicalPlanRef,
         right: LogicalPlanRef,
-        left_on: Vec<ExprRef>,
+        left_on: &[ExprRef],
         right_on: Vec<ExprRef>,
```

Comment:
why does `left_on` take a borrowed `&[ExprRef]` but `right_on` takes an owned `Vec<ExprRef>`

Reply:
It's a little ugly but it's because right and right_on may be used to create new nodes or returned without modification, whereas left and left_on are just referenced for getting the merged columns. So by doing this, I can move instead of copy the right side while not copying the left side.

An alternative maybe is returning all of them (left, left_on, right, right_on). However we'll never be modifying left or left_on in this function. What are your thoughts on this?

Reply:
Decided to change this to return left and left_on too